### PR TITLE
Add Binary Data Type

### PR DESCRIPTION
**Pull request by howhelan**
Adding the binary data type.  BinaryData is stored as a serialized json object in the database, which has two fields:
- `binary_type`: The mime-type of the stored binary file (image or video)
- `binary`: The binary file as a base64 encoded string

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **howhelan**
You can assign the PR to them by writing `/assign @howhelan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-crypto).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-crypto/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/data.rs
Comment (ajpauwels) [inline src/data.rs:63]: I don't think this is an appropriate data type for the binary content. I believe a string HAS to be UTF-8 bytes, I'm surprised this works at all actually. I think a Vec<u8> would be more appropriate.

File: src/data.rs
Comment (ajpauwels) [inline src/data.rs:395]: This goes along with the review comment re using a String as the data type (unless there's something I'm missing here). Should just be able to keep `binary_data_bytes.to_vec()` and remove the wrapping `from_utf8`

**Review by ajpauwels — CHANGES_REQUESTED** [review]

**Pull request merged**
